### ISSUE TITLE
TestUnreservePlugin is flaky

### ISSUE THREAD

**Issue by liggitt** [kubernetes/kubernetes#77573]
This test just merged in #77457 and is flaky. Example failure in https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/77564/pull-kubernetes-integration/1125927448479797249/

```
framework_test.go:310: test #1: Expected the unreserve plugin to be called only once.
```

/sig scheduling
/cc @bsalamat @danielqsj

**Comment by danielqsj** [comment]
It's weird. It was passed yesterday: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/77457/pull-kubernetes-integration/1125654742274412544/

**Comment by tedyu** [comment]
```
					t.Errorf("test #%v: Expected the unreserve plugin to be called only once.", i)
```
Should the value of unresPlugin.numUnreserveCalled be included in the error message ?
That would give us some idea how many times Unreserve is called.

**Comment by danielqsj** [comment]
@tedyu  yep. Added it in https://github.com/kubernetes/kubernetes/pull/77574. Let's see what happened.

**Comment by tedyu** [comment]
When I ran TestUnreservePlugin alone, sometimes it failed.

I doubt moving the reset() call would fix the flaky test.

**Comment by logicalhan** [comment]
Can we revert the test to unblock CI? It seems to be failing pretty consistently at this point (at least for me).

**Comment by tedyu** [comment]
I would propose the following fix:
```
diff --git a/test/integration/scheduler/framework_test.go b/test/integration/scheduler/framework_test.go
index 9ab6f2f7aa..3d0293967a 100644
--- a/test/integration/scheduler/framework_test.go
+++ b/test/integration/scheduler/framework_test.go
@@ -89,6 +89,10 @@ func (pp *PrebindPlugin) Name() string {
        return prebindPluginName
 }

+func (pp *PrebindPlugin) reset() {
+       pp.numPrebindCalled = 0
+}
+
 // Prebind is a test function that returns (true, nil) or errors for testing.
 func (pp *PrebindPlugin) Prebind(pc *framework.PluginContext, pod *v1.Pod, nodeName string) *framework.Status {
        pp.numPrebindCalled++
@@ -306,16 +310,18 @@ func TestUnreservePlugin(t *testing.T) {
                        if err = wait.Poll(10*time.Millisecond, 30*time.Second, podSchedulingError(cs, pod.Namespace, pod.Name)); err != nil {
                                t.Errorf("test #%v: Expected a scheduling error, but didn't get it. error: %v", i, err)
                        }
-                       if unresPlugin.numUnreserveCalled != 1 {
-                               t.Errorf("test #%v: Expected the unreserve plugin to be called only once.", i)
+                       if unresPlugin.numUnreserveCalled != pbdPlugin.numPrebindCalled {
+                               t.Errorf("test #%v: Expected the unreserve plugin to be called %v. but was %v", i,
+                                       pbdPlugin.numPrebindCalled, unresPlugin.numUnreserveCalled)
                        }
                } else {
                        if test.prebindReject {
                                if err = waitForPodUnschedulable(cs, pod); err != nil {
                                        t.Errorf("test #%v: Didn't expected the pod to be scheduled. error: %v", i, err)
                                }
-                               if unresPlugin.numUnreserveCalled != 1 {
-                                       t.Errorf("test #%v: Expected the unreserve plugin to be called only once.", i)
+                               if unresPlugin.numUnreserveCalled != pbdPlugin.numPrebindCalled {
+                                       t.Errorf("test #%v: Expected the unreserve plugin to be called %v. but was %v", i,
+                                               pbdPlugin.numPrebindCalled, unresPlugin.numUnreserveCalled)
                                }
                        } else {
                                if err = waitForPodToSchedule(cs, pod); err != nil {
@@ -327,6 +333,7 @@ func TestUnreservePlugin(t *testing.T) {
                        }
                }
                unresPlugin.reset()
+               pbdPlugin.reset()
                cleanupPods(cs, t, []*v1.Pod{pod})
        }
 }
```

**Comment by tedyu** [comment]
I ran with the above fix several times where the test passed.

**Comment by liggitt** [comment]
Yes, please revert as a first step to unblock others. It can be reintroduced once it's been proved stable in a PR

**Comment by logicalhan** [comment]
#77577 to revert.

**Comment by danielqsj** [comment]
@tedyu  great thanks. It works. #77574 is updated.
Due to it will be reverted at first, I will open a new PR for review.
